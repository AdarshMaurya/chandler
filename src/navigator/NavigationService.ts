import {NavigationActions, StackActions} from 'react-navigation';

let navigator;

export const setTopLevelNavigator = navigatorRef => {
  navigator = navigatorRef;
};

export const navigate = (routeName: string, params?: any) => {
  navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
};

export const pop = () => {
  navigator.dispatch(StackActions.pop());
};

export type INavigateAction = typeof navigate;
export type IPopAction = typeof pop;
