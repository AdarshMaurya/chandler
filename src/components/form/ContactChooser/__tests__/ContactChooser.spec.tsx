import {navigate} from '@navigator/NavigationService';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {ContactChooser} from '../ContactChooser';

jest.mock('@navigator/NavigationService', () => ({
  navigate: jest.fn(),
}));

const navigateMock = navigate as jest.Mock;

describe('Components', () => {
  describe('ContactChooser', () => {
    const onValueChange = jest.fn();
    const contact = {first_name: 'plouf', last_name: 'nathie'};

    it('should renders correctly', () => {
      expect(
        toJson(
          shallow(
            <ContactChooser
              onValueChange={onValueChange}
              value={contact as any}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should renders without contact', () => {
      expect(
        toJson(
          shallow(<ContactChooser onValueChange={onValueChange} value={null} />)
        )
      ).toMatchSnapshot();
    });

    it('should renders with an error', () => {
      expect(
        toJson(
          shallow(
            <ContactChooser
              onValueChange={onValueChange}
              value={null}
              touched={true}
              error="dqwdqw"
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should navigate to ChooseContacts', () => {
      const tree = shallow(
        <ContactChooser onValueChange={onValueChange} value={null} />
      );
      tree.instance().returnData = 'a';
      tree.instance().navigate();
      expect(navigateMock.mock.calls[0]).toEqual([
        'ChooseContacts',
        {returnData: 'a'},
      ]);
    });

    it('returnData should call callback', () => {
      const tree = shallow(
        <ContactChooser onValueChange={onValueChange} value={null} />
      );
      tree.instance().returnData('a');
      expect(onValueChange.mock.calls).toEqual([['a']]);
    });
  });
});
