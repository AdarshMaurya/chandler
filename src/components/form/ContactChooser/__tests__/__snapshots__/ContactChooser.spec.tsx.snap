// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContactChooser should renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <ContactAvatar
    contact={
      Object {
        "first_name": "plouf",
        "last_name": "nathie",
      }
    }
    key="0"
    size={40}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    key="10"
    style={
      Object {
        "flex": 1,
        "marginLeft": 10,
      }
    }
  >
    plouf nathie
  </Text>
  <Touchable
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#325776",
        }
      }
    >
      Edit
    </Text>
  </Touchable>
</Component>
`;

exports[`Components ContactChooser should renders with an error 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      No contact
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
        }
      }
      testID="contact-chooser-error-undefined"
    >
       - 
      dqwdqw
    </Text>
  </Component>
  <Touchable
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#325776",
        }
      }
    >
      Edit
    </Text>
  </Touchable>
</Component>
`;

exports[`Components ContactChooser should renders without contact 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      No contact
    </Text>
  </Component>
  <Touchable
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#325776",
        }
      }
    >
      Edit
    </Text>
  </Touchable>
</Component>
`;
